docs/graphcast/radios/intro.md

```diff
@@ -31,251 +28,196 @@ Once that's done, you can start building your very first Radio.
 ```
 
-:::info
-Don't worry about the environment variables that you see there which we haven't added to the `.env`, they will reside in the `Dockerfile` that we'll create at the end.
-:::
+That line may look intimidating, but really it's just defining a global static instance of a `GraphcastMessage` vector. It is used to save incoming messages after they've been validated, in order to defer their processing for later, because async code is required for the processing but is not allowed inside the handler function.
 
-### Making sure everything works
+Next is an instance of `GossipAgent`:
 
-This next segment will make sure that our operator gets resolved to an on-chain Indexer address and that we're ready to start sending and receiving message via Graphcast.
+```Rust
+pub static GOSSIP_AGENT: OnceCell<GossipAgent> = OnceCell::new();
+```
 
-```typescript
-const indexerAddress = await gossipAgent.init();
+This is also a global static definition, because the Radio handler requires a static immutable context and the handler itself is being passed into the GossipAgent, so it needs to be static as well.
 
-logger.info(`🔦 Radio operator resolved to indexer address ${indexerAddress}`);
+We also need to load our environment variables:
 
-logger.info(`👂 Initialize ping pong Radio with operator status: `, {
-  indexerAddress:
-    indexerAddress ??
-    "Graphcast agent is not registered as an indexer operator",
-  topic: DOMAIN,
-});
+```Rust
+let private_key = env::var("PRIVATE_KEY").expect("No operator private key provided.");
+let eth_node = env::var("ETH_NODE").expect("No ETH URL provided.");
 ```
 
-### Sending messages
-
-Next, we will define a helper function (still inside of the `run` function) which will take care of sending messages to the network.
+Using the Ethereum node we can now set up a provider instance that will help us read on-chain data:
 
-```typescript
-const sendMessage = async (radioPayload) => {
-  const provider = gossipAgent.clientManager.ethClient.provider;
+```Rust
+let provider: Provider<Http> = Provider::<Http>::try_from(eth_node.clone()).unwrap();
+```
 
-  const block = await provider.getBlockNumber();
-  const blockObject = await provider.getBlock(block);
+Next, we will set our Radio name (can be any string) and instantiate a `GossipAgent`:
+
+```Rust
+let radio_name: &str = "ping-pong";
+
+let gossip_agent = GossipAgent::new(
+    private_key,
+    eth_node,
+    radio_name,
+    Some(vec!["ping-pong-content-topic"]),
+    None,
+    None
+)
+.await
+.unwrap();
+```
 
-  logger.info("Sending message with payload: ", radioPayload);
+It's important to note that the `Some(vec!["ping-pong-content-topic"])` can be any vector of strings, but we can also pass in `None`, that way the SDK will default to using the ipfs hashes of the subgraphs that the Indexer is allocation to as initial content topics. But in this case we are overriding it with something much more simple.
```

Comment:
Hmm, I would suggest to say something along the lines of
`Gossip agent takes in an optional vector for content topics. Here we explicitly provide a singleton vector of "ping-pong-content-topic", but you can define topics based on the radio's use case needs. If you leave the field as None, then the agent will automatically fetch your indexer's active allocations and create a list of topics in the format of "radio application name" + the allocated subgraph deployments' IPFS hash.`